ddr/lib/ddr-scanner/dwarf/DwarfParser.cpp

```diff
@@ -117,9 +117,15 @@ dwarf_init(int fd,
 	 * newer compiler versions, so we prefer the more modern dwarfdump.
 	 */
 	char *toolpath = NULL;
+#if defined(AIXPPC)
+	// AIX with OpenXL
+	if (findTool(&toolpath, "which llvm-dwarfdump 2>/dev/null"))
+#else
+	// macOS
 	if (findTool(&toolpath, "xcrun -f dwarfdump 2>/dev/null")
-	||  findTool(&toolpath, "xcrun -f dwarfdump-classic 2>/dev/null")
-	) {
+	||  findTool(&toolpath, "xcrun -f dwarfdump-classic 2>/dev/null"))
+#endif /* defined (AIXPPC) */
+	{
```

Comment:
Suggest extracting the common elements (`if (` at the beginning, `) {` at the end), adding missing `#else` comment:
```c
	if (
#if defined(AIXPPC)
		// AIX with OpenXL
		findTool(&toolpath, "which llvm-dwarfdump 2>/dev/null")
#else /* defined (AIXPPC) */
		// macOS
		findTool(&toolpath, "xcrun -f dwarfdump 2>/dev/null")
	||  findTool(&toolpath, "xcrun -f dwarfdump-classic 2>/dev/null")
#endif /* defined (AIXPPC) */
	) {
```